prometheus/prometheus.go

```diff
@@ -13,19 +13,90 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+var (
+	// DefaultPrometheusOpts is the default set of options used when creating a
+	// PrometheusSink.
+	DefaultPrometheusOpts = PrometheusOpts{
+		Expiration: 60 * time.Second,
+	}
+)
+
+// PrometheusOpts is used to configure the Prometheus Sink
+type PrometheusOpts struct {
+	// Expiration is the duration a metric is valid for, after which it will be
+	// untracked. If the value is zero, a metric is never expired.
+	Expiration time.Duration
```

Comment:
I'd probably make this opt-in, but I can see why you'd do this to make it avoid memory leaks.